src/lib/mongoBackend/mongoUnsubscribeContext.cpp

```diff
@@ -127,19 +127,21 @@ HttpStatusCode mongoUnsubscribeContext
   //
   // Removing subscription from mongo subscription cache
   //
-  LM_T(LmtSubCache, ("removing subscription '%s' (tenant '%s') from mongo subscription cache",
-                     requestP->subscriptionId.get().c_str(),
-                     tenant.c_str()));
+  if (!noCache)
+  {
+    LM_T(LmtSubCache, ("removing subscription '%s' (tenant '%s') from mongo subscription cache",
+                       requestP->subscriptionId.get().c_str(),
+                       tenant.c_str()));
 
-  cacheSemTake(__FUNCTION__, "Removing subscription from cache");
+    cacheSemTake(__FUNCTION__, "Removing subscription from cache");
 
-  CachedSubscription* cSubP = subCacheItemLookup(tenant.c_str(), requestP->subscriptionId.get().c_str());
+    CachedSubscription* cSubP = subCacheItemLookup(tenant.c_str(), requestP->subscriptionId.get().c_str());
 
-  if (cSubP != NULL)
-  {
-    subCacheItemRemove(cSubP);
+    if (cSubP != NULL)
+    {
+      subCacheItemRemove(cSubP);
+    }
   }
-
   cacheSemGive(__FUNCTION__, "Removing subscription from cache");
```

Comment:
cacheSemGive() should be covered inside the `if (!noCache)` block.

Reply:
Fixed in https://github.com/telefonicaid/fiware-orion/pull/3993/commits/52883254a5a505645d96ab1c31d5b6c00b962b7a